### PR TITLE
Added CRAM to alara.py

### PR DESCRIPTION
**Pull request by marissachanlatte**
I added CRAM to alara. I have done some testing myself (results are below). The exponential method seems to be good, but there are a few issues with the last test. 

Testing:

""" Testing the exponentiation part of CRAM. The rat_apprx functions give n(t) = (e^At)*n_0, setting t to 1.0 and 
n_0 to the identity gives e^A """
# Test 1, I = the identity matrix, e^I

IN: 
A = np.eye(3)
print "CRAM: "
print rat_apprx_16(A, 1.0, np.eye(3))
print "e^I: "
print np.e \* np.eye(3)

OUT: 
CRAM: 
[[ 2.71828183  0.          0.        ]
 [ 0.          2.71828183  0.        ]
 [ 0.          0.          2.71828183]]
e^I: 
[[ 2.71828183  0.          0.        ]
 [ 0.          2.71828183  0.        ]
 [ 0.          0.          2.71828183]]
# Test 2, random 3x3 matrix compared to Pade approximation and Taylor

IN:
A = np.random.rand(3, 3)
CRAM_test2 = rat_apprx_16(A, 1.0, np.eye(3))
Pade_test2 = scipy.linalg.expm(A)
Taylor_test2 = scipy.linalg.expm3(A)
print "CRAM: "
print CRAM_test2
print "Pade: "
print Pade_test2
print "Taylor: "
print Taylor_test2
print "Errors: "
print np.linalg.norm(CRAM_test2 - Pade_test2)
print np.linalg.norm(CRAM_test2 - Taylor_test2)

OUT:
CRAM: 
[[ 1.47736655  0.65559436  0.22226281]
 [ 1.85166701  3.1315016   0.7647304 ]
 [ 1.35744556  1.71593956  1.57244038]]
Pade: 
[[ 1.47736655  0.65559437  0.22226282]
 [ 1.85166704  3.13150163  0.76473041]
 [ 1.35744558  1.71593958  1.57244038]]
Taylor: 
[[ 1.47736655  0.65559437  0.22226282]
 [ 1.85166704  3.13150163  0.76473041]
 [ 1.35744558  1.71593958  1.57244038]]
Errors: 
4.85973599949e-08
4.85973601684e-08
# Test 3 Carbon14 decay
# PyNE decay

IN:
from pyne.material import Material
c14_0 = Material({'C14': 1.0})
c14_hl = data.half_life('C14')
c14_1 = c14_0.decay(8*c14_hl)
print(c14_1)

OUT:
Material:
mass = 0.99998805078
density = -1.0
## atoms per molecule = 1.0

C14    0.00390629531964
N14    0.99609370468
# CRAM

IN: CRAM(['C14', 'N14'], c14_hl \* 8.0, [1.0, 0], 16)

OUT: array([ 0.00390625,  0.99609375])
# Test 4 Thorium 232 Decay
# PyNE decay

IN:
th232_0 = Material({'Th232' : 1.0})
th232_hl = data.half_life('Th232')
th232_1 = th232_0.decay(th232_hl)
print th232_1

OUT:
Material:
mass = 0.948152030025
density = -1.0
## atoms per molecule = 1.00000000029

Hg206  4.01984011444e-34
Tl206  1.4307557741e-32
Tl208  7.04340618772e-17
Pb206  1.87243515228e-11
Pb208  0.472658433937
Pb210  3.02694341042e-20
Pb212  4.17699719613e-14
Bi210  1.87099228895e-23
Bi212  3.96172387083e-15
Po210  5.16558236955e-22
Po212  2.08868049467e-25
Po216  1.61111466519e-19
Rn220  6.2924652605e-17
Ra224  3.64403721206e-13
Ra228  2.12846560755e-10
Ac228  2.59700242554e-14
Th228  7.07605987179e-11
Th232  0.52734156576
# CRAM (note values for Pb208 and Pb206, among others, are off)

IN:
nuclides = ['Th232', 'Th228', 'Ac228', 'Ra228', 'Ra224', 'Rn220', 
            'Po216', 'Po212', 'Po210', 'Bi212', 'Bi210', 'Pb212',
            'Pb210', 'Pb208', 'Pb206', 'Tl208', 'Tl206', 'Hg206']
n_0 = np.zeros((18))
n_0[0] = 1.0
CRAM(nuclides, th232_hl, n_0, 16)

OUT:
array([  5.00000000e-01,   6.82714286e-11,   2.50562237e-14,
         2.05357143e-10,   3.57876210e-13,   6.29234887e-17,
         1.64099026e-19,   3.38383509e-25,   1.35304586e-11,
         4.11152940e-15,   4.90075292e-13,   4.33493042e-14,
         7.92857145e-10,   9.99999999e-01,   1.50000000e+00,
         2.07307997e-16,   5.70657194e-16,   5.64953338e-16])

### CODE REVIEW

File: pyne/alara.py [hunk 1]
```diff
@@ -517,3 +519,149 @@ def irradiation_blocks(material_lib, element_lib, data_library, cooling,
...
+                A[k, i] += decay_const(N_id[k])
+            
+    return A
+
+def _rat_apprx_14(A, t, n_0):
+    """ CRAM of order 14
+
+    Parameters
+    ---------
+
```
Comment (scopatz) [inline pyne/alara.py]: DOC: No empty line here.

File: pyne/alara.py [hunk 2]
```diff
@@ -517,3 +519,149 @@ def irradiation_blocks(material_lib, element_lib, data_library, cooling,
     s += "dump_file {0}\n".format(dump_file)
 
     return s
+
+""" Chebyshev Rational Approximation Method functions
+"""
```
Comment (scopatz) [inline pyne/alara.py]: Docstring needs to be at the top of the file.

File: pyne/alara.py [hunk 3]
```diff
@@ -25,8 +25,10 @@
 
 from pyne.mesh import Mesh, MeshError
 from pyne.material import Material, from_atom_frac
-from pyne.nucname import serpent, alara, znum, anum
-from pyne.data import N_A
+from pyne.nucname import serpent, alara, znum, anum, id
```
Comment (scopatz) [inline pyne/alara.py]: `id()` is a Python builtin function.  Don't import directly.  Do something like:

``` python
from pyne import nucname
nucname.id(...)
```

File: pyne/alara.py [hunk 4]
```diff
@@ -517,3 +519,149 @@ def irradiation_blocks(material_lib, element_lib, data_library, cooling,
...
+    n = 0*n_0
+
+    for j in range(8):
+        n = n + np.dot(alpha[j]*n_0, np.linalg.inv(A - theta[j] * np.identity(np.shape(A)[0])))
+
+    n = 2*n.real
+    n = n + alpha_0*n_0
+    return n
+
+def CRAM(N, t, n_0, order):
```
Comment (scopatz) [inline pyne/alara.py]: PEP8: function names should be `lowercase_with_underscores()`

File: pyne/alara.py [hunk 5]
```diff
@@ -517,3 +519,149 @@ def irradiation_blocks(material_lib, element_lib, data_library, cooling,
...
+	Order of method. Only 14 and 16 are supported.
+    """
+
+    n_0 = np.array(n_0)
+    A = build_matrix(N)
+    
+    if order == 14:
+        return rat_apprx_14(A, t, n_0)
+    
+    if order == 16:
```
Comment (scopatz) [inline pyne/alara.py]: This should be an `elif`

File: pyne/alara.py [hunk 6]
```diff
@@ -517,3 +519,149 @@ def irradiation_blocks(material_lib, element_lib, data_library, cooling,
...
+        return rat_apprx_16(A, t, n_0)
+    
+    else:
+        return 'Rational approximation of degree %d is not supported.' % order
```
Comment (scopatz) [inline pyne/alara.py]: This should raise a ValueError with this message.  Also don't use `%` for string formatting.  I would rewrite this as:

``` python
msg = 'Rational approximation of degree {0} is not supported.'.format(order)
raise ValueError(msg)
```

File: pyne/alara.py [hunk 7]
```diff
@@ -517,3 +519,149 @@ def irradiation_blocks(material_lib, element_lib, data_library, cooling,
...
+    A : numpy array
+	Burnup matrix
+    t : float
+	Time step
+    n_0: numpy array
+	Inital composition vector
+    """
+	
+    theta = np.array([ complex(-8.8977731864688888199, 16.630982619902085304), 
+                   complex(-3.7032750494234480603, 13.656371871483268171),
```
Comment (scopatz) [inline pyne/alara.py]: It is probably better to use complex literals, eg

``` python
-3.7032750494234480603 + 13.656371871483268171j
```

Notice the `j` at the end.  Silly electrical engineers.

File: pyne/alara.py [hunk 8]
```diff
@@ -517,3 +519,149 @@ def irradiation_blocks(material_lib, element_lib, data_library, cooling,
...
+    
+    alpha = np.array([ complex( -.000071542880635890672853, .00014361043349541300111), 
+                   complex(.0094390253107361688779, -.01784791958483017511), 
+                   complex(-.37636003878226968717, .33518347029450104214), 
+                   complex(-23.498232091082701191, -5.8083591297142074004), 
+                   complex(46.933274488831293047, 45.643649768827760791), 
+                   complex(-27.875161940145646468, -102.14733999056451434), 
+                   complex(4.8071120988325088907, -1.3209793837428723881)])
+    
+    alpha_0 = np.array([1.8321743782540412751*10**(-14)])
```
Comment (scopatz) [inline pyne/alara.py]: Much better to use exponential notation: `1.8321743782540412751e-14`

**Comment by scopatz** [comment]
Thanks for putting this in @mzweig! Can you add the tests to the test suite.  These live in the `tests/` directory.  Let me know if you have any questions about how to do this.

**Comment by rachelslaybaugh** [comment]
@mzweig did you mean to close this? Can you create a new one if this one is really supposed to be closed for some reason? No one can review this work if there's no PR.

**Comment by marissachanlatte** [comment]
@rachelslaybaugh This got closed on accident when I was setting up a new build on my new machine. I was waiting to create a new PR until I got the test running (I'm still having some issues updating my build), but I can make one without the test right now if you'd like.

**Comment by rachelslaybaugh** [comment]
No, waiting until it's ready to evaluate is fine.

**Pull request closed without merging**